akka-stream/src/main/scala/akka/stream/scaladsl/Source.scala

```diff
@@ -677,8 +677,7 @@ object Source {
    * If the content is [[akka.stream.CompletionStrategy.immediately]] the completion will be signaled immediately.
    * Otherwise, if the content is [[akka.stream.CompletionStrategy.draining]] (or anything else)
    * already buffered elements will be sent out before signaling completion.
-   * Sending [[akka.actor.PoisonPill]] will signal completion immediately but this behavior is deprecated and scheduled to be removed.
-   * Using [[akka.actor.ActorSystem.stop]] to stop the actor and complete the stream is *not supported*.
+   * Using [[akka.actor.PoisonPill]] or [[akka.actor.ActorSystem.stop]] to stop the actor and complete the stream is *not supported*.
```

Comment:
It's somewhat scary to change behavior in a patch release even though we have had it deprecated for 1 year. What will happen if PoisonPill is still used?

Reply:
Without testing I'd expect the stream actor to terminate, but the stream to stay open.

Reply:
Note to ourselves if we decide we shouldn't: we probably shouldn't mark an issue as triaged and assign it to a milestone if we don't think it should be done until the next minor release. ;)

Reply:
This might be better suited for Milestone 2.7 or 3.0

Reply:
We can absolutely do it in 2.7.0. That's when we normally remove deprecated methods and change configuration.

I would like that we think some more about it before we close (defer) the PR. I might be overly conservative. It's more than one year of deprecation warning.

Reply:
The deprecation and warning have been added in Akka 2.5.22 almost 2 years ago. One could argue that if someone didn't notice it till now, delaying to Akka 2.7 will probably not change that.

Reply:
Thanks for that info. Good that someone is on top of things here ;-)
I'd say that we go ahead and merge this after reviews.

Reply:
Just missing that second review ;-)